src/main/java/uk/sky/cqlmigrate/CassandraClusterFactory.java

```diff
@@ -1,29 +1,36 @@
 package uk.sky.cqlmigrate;
 
-import com.datastax.driver.core.Cluster;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.session.Session;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class CassandraClusterFactory {
 
     /**
-     * Creates an instance of cassandra {@link Cluster} based on the provided configuration
+     * Creates an instance of cassandra {@link Session} based on the provided configuration
      *
-     * @param hosts Addresses of the nodes to add as contact points (as described in
-     *              {@link Cluster.Builder#addContactPoint}).
-     * @param port The port to use to connect to the Cassandra hosts.
+     * @param hosts    Addresses of the nodes to add as contact points (as described in
+     *                 {@link CqlSession}).
+     * @param port     The port to use to connect to the Cassandra hosts.
      * @param username the username to use to login to Cassandra hosts.
      * @param password the password corresponding to {@code username}.
-     *
-     * @return a configured Cluster
+     * @return a configured Session
      */
-    public static Cluster createCluster(String[] hosts, int port, String username, String password) {
-        Cluster.Builder builder = Cluster.builder()
-                .addContactPoints(hosts)
-                .withPort(port);
+    public static Session createCluster(String[] hosts, int port, String username, String password) {
+        List<InetSocketAddress> cassandraHosts = Stream.of(hosts)
+                .map(host -> new InetSocketAddress(host, port))
+                .collect(Collectors.toList());
 
         if (username != null && password != null) {
-            builder = builder.withCredentials(username, password);
+            return CqlSession.builder()
+                    .addContactPoints(cassandraHosts)
+                    .withAuthCredentials(username, password).build();
         }
 
-        return builder.build();
+        return CqlSession.builder().addContactPoints(cassandraHosts).build();
```

Comment:
clients might be using it but i feel we should remove it. it doesnt give any value to that specific library, clients can have this factory on their side and since this change is already breaking we can get rid of this one now.

Are you happy to remove it @davidh87 ?

Reply:
it's been removed